### PR TITLE
Improve messaging around sys info for other languages

### PR DESCRIPTION
**Pull request by rchiodo**
For #340

Also fixed some cases in remote where the python extension was being forced. Python extension is not required for remote support.

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/datascience/jupyter/jupyterServerWrapper.ts [hunk 1]
```diff
@@ -53,7 +54,8 @@ type JupyterServerClassType = {
         kernelSelector: KernelSelector,
         interpreterService: IInterpreterService,
         outputChannel: IOutputChannel,
-        progressReporter: ProgressReporter
+        progressReporter: ProgressReporter,
+        exxtensionChecker: IPythonExtensionChecker
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/jupyterServerWrapper.ts:58]: extensionChecker

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/datascience/jupyter/jupyterNotebook.ts [hunk 2]
```diff
@@ -453,32 +453,50 @@ export class JupyterNotebookBase implements INotebook {
...
+                    source: []
+                },
+                id: uuid(),
+                file: '',
+                line: 0,
+                state: CellState.finished
+            };
+        } else {
+            // tslint:disable-next-line: no-suspicious-comment
+            // TODO: Provide a way for other languages to return this information
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/jupyterNotebook.ts:486]: **We already have all of the information we need:**
The kernelSpec info is what contains this information.
This is the first request to the Jupyter kernel by the Jlab API.

**Here's the content for Julia**
Screen Shot 2020-11-13 at 16 36 50

**Here's the content for Java**
```json
    {'text': 'BeakerX Home', 'url': 'http://BeakerX.com'}, 
	{'text': 'File an Issue', 'url': 'https://github.com/twosigma/beakerx/issues/new'}, 
    {'text': 'Two Sigma Open Source', 'url': 'http://opensource.twosigma.com/'}, 
    {'text': 'BeakerX JavaDoc', 'url': 'javadoc/index.html'}
], 
	'implementation': 'java', 
    'banner': 'BeakerX 1.4.0\nJava 11.0.8\nhash f8c52c4\nbuild time 2019-03-13 19:28 EDT',
	'beakerx': True, 
	'implementation_version': 'BeakerX 1.4.0\n%s %s\nhash f8c52c4\nbuild time 2019-03-13 19:28 EDT'
```

**Looks like the banner is what we need to display**
The version information is already displayed in the kernel.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
As this is a PR to ensure we support other kernels (python agnostic), i'm blocking so that we display the kernel banner received.
I'd suggest we do the same for Python as well (on top of what we already display).
We don't need an extra request. I found that jlab already does this.

I made an explicity call in Native Notebooks, however the kernel already has a property to expose this `Kernel.info: KernelMessage.IInfoReply | null`.

Hence we can use that as is.

File: src/client/datascience/jupyter/jupyterNotebook.ts (hunk 2 above)
Comment (rchiodo) [reply src/client/datascience/jupyter/jupyterNotebook.ts:486]: That does sound better. I'll try that and see what happens.

Comment (rchiodo) [reply src/client/datascience/jupyter/jupyterNotebook.ts:486]: Yeah not sure. It works pretty good for python. Julia is weird:

image

First banner is from julia.

Second from the python kernel.

Comment (rchiodo) [reply src/client/datascience/jupyter/jupyterNotebook.ts:486]: Maybe I should include the help links too

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts:486]: Do we need the message `Start xyz kernel`?
One less line, considering the fact that you're planning on adding the help lnks as well.

Comment (rchiodo) [reply src/client/datascience/jupyter/jupyterNotebook.ts:486]: That's too wordy I think

image

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts:486]: Lol, agreed.

Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts]: I think this would be something that can be displayed on the left hand side (jupyter panel).
Nice place, pity we don't expose the kernel to extension authors, if we did, then an extension could display a WebView in the sidebar with & all the links would light up... 

😉  - exposing jupyter lab kernel api to extension authors, something we briefly touched on

Comment (rchiodo) [reply src/client/datascience/jupyter/jupyterNotebook.ts]: Yeah I think we can do a lot if we expose kernels/interactive window execution/ cell ranges.

**Review by DonJayamanne — APPROVED** [review]

File: src/client/datascience/jupyter/jupyterNotebook.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/jupyterNotebook.ts]: looking forward to that day, got so many ideas... want to light up this Jupyter panel & make it look like a christmas tree.. plenty of bells and whistles.. 😄

**Review by DavidKutu — APPROVED** [review]
